libs/execution_context/ebpf_core.c

```diff
@@ -2321,6 +2321,11 @@ _ebpf_core_trace_printk(_In_reads_(fmt_size) const char* fmt, size_t fmt_size, i
         return -1;
     }
 
+    // If the provider is not enabled, don't bother with the rest.
+    if (!TraceLoggingProviderEnabled(ebpf_tracelog_provider, EBPF_TRACELOG_LEVEL_INFO, EBPF_TRACELOG_KEYWORD_PRINTK)) {
```

Comment:
qq - does the EBPF_TRACELOG_LEVEL_INFO ensure that more verbose levels are also accounted for in this check?

Reply:
Yes and this is the same check that is made after all the formatting is done, so there is no observable difference in behavior (besides not wasting CPU).